### ISSUE TITLE
Log number of API requests for grant reporting

### ISSUE THREAD

**Issue by dboyer** [open-apparel-registry/open-apparel-registry#472]
## Overview

We need to be able to report the total number of API calls within a certain time period.

### Is your feature request related to a problem? Please describe.

This metric is currently not tracked and we need to have the information for reporting to the funder. 

### Describe the solution you'd like

~~Administrators have access to a tool in the Django admin that supports viewing the number of API calls between two dates. ~~
Build a developer-facing tool that can report the number of non-application API calls made between a pair of dates. See the issue discussion on how this might be accomplished via an Athena query of the access logs.

**Comment by hectcastro** [comment]
This solution wouldn't easily support a Django admin level solution (although, may it cool with Boto3?), but I wanted to surface that we persist all of the HTTP access logs in S3. There are tables in AWS Athena to [query](https://eu-west-1.console.aws.amazon.com/athena/home?region=eu-west-1#query/history/45312b0e-95b0-4fbd-9b40-82c25c937ad6) them via SQL to get an API call count:

```sql
SELECT count(*)
FROM "default"."prd_cloudfront_logs"
WHERE uri LIKE '/api/%'
        AND referrer <> 'http://checks.panopta.com/';
```

**Comment by jwalgran** [comment]
The Athena based solution may be the way to go. Our stakeholders don't need ad-hoc access to the metrics, so an internal command we can run monthly or quarterly is all we need.

Our application uses the API, but adding another `referrer <> 'https://openapparel.org/'` clause to the where should get us the number of non-web app consumers of the API.

**Comment by jwalgran** [comment]
I just tried Athena and it is very powerful. Thanks for the tip.

Here is a modification I made to look for requests not originating from the web app

@hectcastro what is our current retention setting? How far back can we go?

**Comment by hectcastro** [comment]
> @hectcastro what is our current retention setting? How far back can we go?

There are currently no retention settings. If they existed, they'd be at the underlying S3 bucket level, which in this case is `openapparelregistry-production-logs-eu-west-1`.

**Comment by jwalgran** [comment]
Implemented in #644 and #646
